test/rubygems/test_gem_specification.rb

```diff
@@ -2983,16 +2983,29 @@ def test_validate_license_with_invalid_exception
     util_setup_validate
 
     use_ui @ui do
-      @a1.licenses = ['GPL-2.0+ WITH Autocofn-exception-2.0']
+      @a1.licenses = ['MIT WITH Autocofn-exception-2.0']
       @a1.validate
     end
 
     assert_match <<-warning, @ui.error
-WARNING:  license value 'GPL-2.0+ WITH Autocofn-exception-2.0' is invalid.  Use a license identifier from
+WARNING:  license value 'MIT WITH Autocofn-exception-2.0' is invalid.  Use a license identifier from
```

Comment:
I think we should use `GPL-2.0-or-later WITH Autocofn-exception-2.0` here (and a few lines above), because the extention text only makes sense for a GPL license.  Future SPDX metadata may allow detecting license/exception mismatches automatically.  We aren't there yet, but best to stay forward-compatible.

And it may be worth changing "Use a license identifier from..." -> "use identifiers from..." to avoid folks interpreting this as a license-ID issue (in this case, it's an exception-ID issue).

Reply:
I avoided using `GPL-2.0-or-later WITH Autocofn-exception-2.0` because `GPL-3.0-with-autoconf-exception` would be suggested as an alternative for the invalid license. Now, I noticed that deprecated license `GPL-3.0-with-autoconf-exception` being suggested is not favorable, and fixed it in f81e6b2.